### PR TITLE
[tool] Add note about clang version used in CI to tool readme

### PR DESCRIPTION
**Pull request by Rexios80**
This repo uses clang 15 to format files. Newer versions of clang format code differently. This PR adds a note to the tool README about this and adds sample instructions for how to install clang 15 on macOS. We might want to add instructions for other platforms.

Fixes https://github.com/flutter/flutter/issues/152651

Additional context
- https://github.com/flutter/packages/pull/7267
- https://github.com/flutter/packages/pull/7271

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

### CODE REVIEW

**Comment by Rexios80** [comment]
@stuartmorgan It appears that Fuchsia is using clang version `16.0.0` now. Do you think it is important to keep the version this repo uses in sync with that? Or should we leave it at `15.0.0`?

**Comment by stuartmorgan** [comment]
> @stuartmorgan It appears that Fuchsia is using clang version `16.0.0` now. Do you think it is important to keep the version this repo uses in sync with that?

I can't think of any reason why they would need to match.

File: script/tool/README.md [hunk 1]
```diff
@@ -47,6 +47,14 @@ anywhere within a package).
 dart run script/tool/bin/flutter_plugin_tools.dart format --packages package_name
 ```
 
+The `flutter/packages` repository uses clang version `15.0.0` . Newer versions of clang may format code differently. Here is an example of how to install clang version 15 on macOS using [Homebrew](https://brew.sh):
+
+```zsh
+brew tap llvm-hs/homebrew-llvm
```
[the remark this answers is not in the document]
Comment (Rexios80) [reply script/tool/README.md:53]: It looks like it's just maintained by them, and not specific to Haskell. The more "official" looking way to install `clang-format` with homebrew (https://formulae.brew.sh/formula/clang-format) doesn't support version 15.

File: script/tool/README.md [hunk 2]
```diff
@@ -47,6 +47,14 @@ anywhere within a package).
...
+brew install llvm@15
+echo 'export PATH="/opt/homebrew/opt/llvm@15/bin:$PATH"' >> ~/.zshrc
```
[the remark this answers is not in the document]
Comment (Rexios80) [reply script/tool/README.md:55]: Probably. The message above does say this is an example and not the definitive installation method. If you want to remove it we can. I tested it myself and it works.

**Comment by Rexios80** [comment]
I agree that maintaining installation instructions for clang is outside the scope of this repo. I removed the macOS example.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**